File: app/src/main/java/ru/job4j/todolist/StoreContentProvider.java
package ru.job4j.todolist;

import android.content.ContentProvider;
import android.content.ContentValues;
import android.database.Cursor;
import android.net.Uri;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Objects;
import java.util.stream.Collectors;

public class StoreContentProvider extends ContentProvider {

    public static final Uri CONTENT_URI = Uri.parse("content://ru.job4j.todo/items");

    private final Store store = Store.getInstance();

    @Override
    public boolean onCreate() {
        return true;
    }

    @Nullable
    @Override
    public Cursor query(@NonNull Uri uri, @Nullable String[] projection, @Nullable String selection, @Nullable String[] selectionArgs, @Nullable String sortOrder) {
        return new StoreCursor(new Store(store.getTasks().stream()
                .filter(task -> task.getName().contains(Objects.requireNonNull(selection)))
                .collect(Collectors.toList())));
    }

    @Nullable
    @Override
    public String getType(@NonNull Uri uri) {
        return null;
    }

    @Nullable
    @Override
    public Uri insert(@NonNull Uri uri, @Nullable ContentValues values) {
        return null;
    }

    @Override
    public int delete(@NonNull Uri uri, @Nullable String selection, @Nullable String[] selectionArgs) {
        return 0;
    }

    @Override
    public int update(@NonNull Uri uri, @Nullable ContentValues values, @Nullable String selection, @Nullable String[] selectionArgs) {
        return 0;
    }
}

File: app/src/main/java/ru/job4j/todolist/contentProvider/StoreContentProvider.kt
package ru.job4j.todolist.contentProvider

import android.content.ContentProvider
import android.content.ContentValues
import android.database.Cursor
import android.net.Uri
import ru.job4j.todolist.store.SqlStore

class StoreContentProvider : ContentProvider() {
    private var store: SqlStore? = null

    companion object {
        val CONTENT_URI = Uri.parse("content://ru.job4j.todo/items")
    }

    override fun onCreate(): Boolean {
        store = SqlStore(context!!)
        return true
    }

    override fun query(uri: Uri, projection: Array<String>?, selection: String?, selectionArgs: Array<String>?, sortOrder: String?): Cursor? =
        StoreCursor(store!!.getFilteredTasks(selection!!))

    override fun getType(uri: Uri): String? = null

    override fun insert(uri: Uri, values: ContentValues?): Uri? = null

    override fun delete(uri: Uri, selection: String?, selectionArgs: Array<String>?): Int = 0

    override fun update(uri: Uri, values: ContentValues?, selection: String?, selectionArgs: Array<String>?): Int = 0
}